### PR TITLE
Migrate the docker client code from dockertools to dockershim

### PR DESCRIPTION
**Pull request by yujuhong**
Move docker client code from dockertools to dockershim/libdocker. This includes
DockerInterface (renamed to Interface), FakeDockerClient, etc.

This is part of #43234

### CODE REVIEW

**Comment by yujuhong** [comment]
@k8s-bot unit test this

**Comment by yujuhong** [comment]
k	@k8s-bot unit test this

**Comment by yujuhong** [comment]
Rebased. PTAL!

**Comment by yujuhong** [comment]
@k8s-bot kubemark e2e test this

**Comment by feiskyer** [comment]
@k8s-bot kubemark e2e test this

**Comment by yujuhong** [comment]
@k8s-bot bazel test this

**Comment by yujuhong** [comment]
@k8s-bot verify test this

**Comment by dchen1107** [comment]
/approve 

I will leave @Random-Liu to review this pr in detail. Thanks!

File: pkg/kubelet/sysctl/runtime.go [hunk 1]
```diff
@@ -31,8 +30,13 @@ const (
 	// (e.g., 1.24). Append the version with a ".0" so that it works
 	// with both the CRI and dockertools comparison logic.
 	dockerMinimumAPIVersion = "1.24.0"
+
+	dockerTypeName = "docker"
 )
 
+// TODO: The admission logic in this file is runtime-dependent. It should be
+// changed to be generic and CRI-compatible.
+
```
Comment (Random-Liu) [inline pkg/kubelet/sysctl/runtime.go]: Extra empty line

**Review by Random-Liu — COMMENTED** [review]
Except changing `dockertools` to `libdocker`, did you only move things around?

Did you make any actual change? If not, the PR LGTM already. :)

File: pkg/kubelet/sysctl/runtime.go (hunk 1 above)
Comment (yujuhong) [inline pkg/kubelet/sysctl/runtime.go]: The comment was intended for the entire file, not just the struct below. Hence the extra line for separation.

**Comment by yujuhong** [comment]
> Except changing dockertools to libdocker, did you only move things around?
> 
> Did you make any actual change? If not, the PR LGTM already. :)

No functional changes. I only made decisions about where the code should reside :-)

File: pkg/kubelet/sysctl/runtime.go (hunk 1 above)
Comment (Random-Liu) [inline pkg/kubelet/sysctl/runtime.go]: SGTM

**Comment by yujuhong** [comment]
Applying lgtm based on @Random-Liu's review.

**Pull request merged**